### PR TITLE
fix: stream return EOF when openai return error

### LINKED ISSUE #173: ChatCompletionRequest with stream: true return EOF
When using ChatCompletionRequest and setting `stream: true` to send multiple conversation content, EOF is returned and no error information can be obtained.

### PR DESCRIPTION
**Pull request by liushuangls**
fix #173

### CODE REVIEW

**Review by sashabaranov — CHANGES_REQUESTED** [review]
Thank you for the PR! 🔥 Looks good overall, but there are a couple of comments. Would you be able to fix those?

File: chat_stream.go [hunk 1]
```diff
@@ -48,12 +51,19 @@ func (stream *ChatCompletionStream) Recv() (response ChatCompletionStreamRespons
...
+		if errRes, _ := stream.errAccumulator.unmarshalError(); errRes != nil {
+			err = fmt.Errorf("error, %w", errRes.Error)
+		}
 		return
 	}
 
 	var headerData = []byte("data: ")
 	line = bytes.TrimSpace(line)
 	if !bytes.HasPrefix(line, headerData) {
+		if writeErr := stream.errAccumulator.write(line); writeErr != nil {
```
Comment (sashabaranov) [inline chat_stream.go:63]: We can cover this with tests if we make `errorAccumulator` an interface and provide failing one like here https://github.com/sashabaranov/go-openai/blob/aa149c1bf8ccd77403e07fdb7b0122f27d4c9b6b/request_builder_test.go#L30

File: unmarshaler.go [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package openai
+
+import (
+	"encoding/json"
+)
+
+type unmarshaler interface {
```
Comment (sashabaranov) [inline unmarshaler.go:7]: Should we use that instead of json.Unmrashal?

**Review by sashabaranov — APPROVED** [review]

**Comment by juggli** [comment]
nice. I meet the same EOF error in v1.5.2.

**Pull request merged**